internal/ingress/controller/kong.go

```diff
@@ -72,10 +93,24 @@ func (n *KongController) OnUpdate(ctx context.Context, state *kongstate.KongStat
 	} else {
 		err = n.onUpdateDBMode(targetContent)
 	}
+	var target []byte
+	if n.cfg.LogSensitiveConfig {
+		target, _ = json.Marshal(targetContent)
+	} else {
+		state.Sanitize()
+		sanitizedContent := n.toDeckContent(ctx, state)
+		target, _ = json.Marshal(sanitizedContent)
+	}
 	if err != nil {
+		if n.cfg.LogLevel == "debug" {
+			diff, _ := getDiff(target, n.lastConfig)
```

Comment:
FWIW, [existing CLI tools can diff two JSONs](https://stackoverflow.com/questions/31930041/using-jq-or-alternative-command-line-tools-to-compare-json-files). Adding extra diffing functionality here may end up being superfluous, unless CLI diffing is a significant drag.

I understand that the UX with a diff produced directly by KIC is more streamlined, but I don't see how it would justify the extra complexity added to the codebase.